src/Illuminate/Database/Eloquent/Concerns/HasEvents.php

```diff
@@ -351,4 +351,27 @@ public static function unsetEventDispatcher()
     {
         static::$dispatcher = null;
     }
+
+    /**
+     * Execute a callback without the event dispatcher.
+     *
+     * @param  callable  $callback
+     * @return mixed
+     */
+    public static function withoutEventDispatcher(callable $callback)
+    {
+        $dispatcher = static::getEventDispatcher();
+
+        static::unsetEventDispatcher();
+
+        try {
+            $result = $callback();
+        } finally {
+            if ($dispatcher) {
+                static::setEventDispatcher($dispatcher);
+            }
+        }
+
+        return $result;
+    }
```

Comment:
Good call on removing the temp variable - I didn't realise `finally` would still be executed after a `return`. Thanks for teaching me something new!

Regarding the `if` condition - the reason I added this is because `setEventDispatcher()` requires a valid `Dispatcher` and it's possible that the user didn't actually have an event dispatcher set. 

You might say that someone shouldn't use this method if they don't have an event dispatcher set, but I guess it would be nice to be able to call the method without having to worry about that.